From deae6c3b57c3919946a5ce1613e667a3240cf158 Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Mon, 15 Apr 2024 12:45:09 +0200
Subject: [PATCH 001/100] introduce pc_rhel_9_5_compat

RH-Author: Paolo Bonzini <pbonzini@redhat.com>
RH-MergeRequest: 245: SEV-SNP support
RH-Jira: RHEL-39544
RH-Acked-by: Thomas Huth <thuth@redhat.com>
RH-Acked-by: Bandan Das <bdas@redhat.com>
RH-Acked-by: Vitaly Kuznetsov <vkuznets@redhat.com>
RH-Commit: [1/91] cfd402fa5080eddba7c954e81ed79f9a1dd654cf (bonzini/rhel-qemu-kvm)

Allow undoing backported changes that impact guest ABI.

Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
---
 hw/i386/pc.c         | 4 ++++
 hw/i386/pc_piix.c    | 2 ++
 hw/i386/pc_q35.c     | 2 ++
 include/hw/i386/pc.h | 3 +++
 4 files changed, 11 insertions(+)

diff --git a/hw/i386/pc.c b/hw/i386/pc.c
index 4a154c1a9a..648762d908 100644
--- a/hw/i386/pc.c
+++ b/hw/i386/pc.c
@@ -348,6 +348,10 @@ GlobalProperty pc_rhel_compat[] = {
 };
 const size_t pc_rhel_compat_len = G_N_ELEMENTS(pc_rhel_compat);
 
+GlobalProperty pc_rhel_9_5_compat[] = {
+};
+const size_t pc_rhel_9_5_compat_len = G_N_ELEMENTS(pc_rhel_9_5_compat);
+
 GlobalProperty pc_rhel_9_3_compat[] = {
     /* pc_rhel_9_3_compat from pc_compat_8_0 */
     { "virtio-mem", "unplugged-inaccessible", "auto" },
diff --git a/hw/i386/pc_piix.c b/hw/i386/pc_piix.c
index 6b260682eb..bef3e8b73e 100644
--- a/hw/i386/pc_piix.c
+++ b/hw/i386/pc_piix.c
@@ -1015,6 +1015,8 @@ static void pc_machine_rhel760_options(MachineClass *m)
     object_class_property_set_description(oc, "x-south-bridge",
                                      "Use a different south bridge than PIIX3");
 
+    compat_props_add(m->compat_props, pc_rhel_9_5_compat,
+		     pc_rhel_9_5_compat_len);
     compat_props_add(m->compat_props, hw_compat_rhel_9_5,
 		     hw_compat_rhel_9_5_len);
     compat_props_add(m->compat_props, hw_compat_rhel_9_4,
diff --git a/hw/i386/pc_q35.c b/hw/i386/pc_q35.c
index 2b54944c0f..9adcdadce8 100644
--- a/hw/i386/pc_q35.c
+++ b/hw/i386/pc_q35.c
@@ -734,6 +734,8 @@ static void pc_q35_machine_rhel940_options(MachineClass *m)
     pcmc->smbios_stream_product = "RHEL";
     pcmc->smbios_stream_version = "9.4.0";
 
+    compat_props_add(m->compat_props, pc_rhel_9_5_compat,
+		     pc_rhel_9_5_compat_len);
     compat_props_add(m->compat_props, hw_compat_rhel_9_5,
 		     hw_compat_rhel_9_5_len);
 }
diff --git a/include/hw/i386/pc.h b/include/hw/i386/pc.h
index a984c951ad..87420783ab 100644
--- a/include/hw/i386/pc.h
+++ b/include/hw/i386/pc.h
@@ -294,6 +294,9 @@ extern const size_t pc_compat_2_0_len;
 extern GlobalProperty pc_rhel_compat[];
 extern const size_t pc_rhel_compat_len;
 
+extern GlobalProperty pc_rhel_9_5_compat[];
+extern const size_t pc_rhel_9_5_compat_len;
+
 extern GlobalProperty pc_rhel_9_3_compat[];
 extern const size_t pc_rhel_9_3_compat_len;
 
-- 
2.39.3

